java-checks/src/main/java/org/sonar/java/checks/UndocumentedApiCheck.java

```diff
@@ -143,14 +130,14 @@ private void visitNode(Tree tree, Tree reportTree, SymbolMetadata symbolMetadata
       if (javadoc.noMainDescription() && !isNonVoidMethodWithNoParameter(tree, javadoc)) {
         context.reportIssue(this, reportTree, "Document this public " + getType(tree) + " by adding an explicit description.");
       } else {
-        List<String> undocumentedParameters = javadoc.undocumentedParameters(tree);
+        Set<String> undocumentedParameters = javadoc.undocumentedParameters();
         if (!undocumentedParameters.isEmpty()) {
           context.reportIssue(this, reportTree, "Document the parameter(s): " + undocumentedParameters.stream().collect(Collectors.joining(", ")));
         }
         if (hasNonVoidReturnType(tree) && javadoc.noReturnDescription()) {
           context.reportIssue(this, reportTree, "Document this method return value.");
         }
-        List<String> undocumentedExceptions = javadoc.undocumentedThrownExceptions(tree);
+        Set<String> undocumentedExceptions = javadoc.undocumentedThrownExceptions();
```

Comment:
I think it should remains a `List`, and not a `Set`. I have the feeling that with a set we could not guarantee order of the results, and therefore have some impacts on tests (flickering fail/pass depending of orders). I would stay with lists to guarantee that.